### PR TITLE
Optimize the build [ECR-534]:

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

- Profile the build with various options (see the results in Jira :arrow_down_small: )
- Add Maven JVM configuration files that improve the build time, according to the experiments
- Make all tests parallel-execution friendly. The remaining issues are:
  - TestkitExtension can't work with `@Nested` tests. 
    Already fixed in master, scheduled to 5.5.
    See https://github.com/junit-team/junit5/issues/1925
  - MockitoExtension does not work in parallel.
- Enable parallel builds and parallel tests on CI. 
  @dip56 have already stress-tested them to be reliable enough.
- Document how this can be enabled locally.

---
See: https://jira.bf.local/browse/ECR-534

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
That's how a testkit failure looks like:

```
[ERROR] Tests run: 12, Failures: 0, Errors: 9, Skipped: 0, Time elapsed: 0.599 s <<< FAILURE! - in com.exonum.binding.qaservice.transactions.ErrorTxIntegrationTest
[ERROR] getValue{TestKit}  Time elapsed: 0.019 s  <<< ERROR!
org.junit.jupiter.api.extension.ParameterResolutionException: 
  Discovered multiple competing ParameterResolvers for parameter
  [com.exonum.binding.testkit.TestKit arg0] in method [void com.exonum.binding.qaservice.QaServiceImplIntegrationTest.getValue(com.exonum.binding.testkit.TestKit)]: 
   - com.exonum.binding.testkit.TestKitExtension@27ac2b5b,
   - com.exonum.binding.testkit.TestKitExtension@61db87a5,
   - com.exonum.binding.testkit.TestKitExtension@68e1cae4
```

**Comment by dmitry-timofeev** [comment]
Submitted junit-team/junit5#1925

**Comment by dmitry-timofeev** [comment]
More experiments, this time, with `mvn install -P ci-build` — which is executed from `./run_all_tests.sh`. See results in the Jira: https://jira.bf.local/browse/ECR-534?focusedCommentId=48422&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-48422

**Comment by dmitry-timofeev** [comment]
Please see the updated description — and review, it must be ready :upside_down_face:

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -117,9 +117,7 @@
         <version>2.22.2</version>
         <configuration>
           <argLine>
-            <!-- TODO: Replace with a path to the libjava_bindings.so to enable
-                 integration tests that require a native library. -->
-            -Djava.library.path=${project.basedir}/core/rust/target/debug
+            -Djava.library.path=${env.EXONUM_HOME}/lib/native
```
Comment (skletsun) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: Is this going to be used for integration tests? If so - shouldn't it point to `target/debug/deps`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -57,6 +57,37 @@ By default, the Exonum Java app is built in debug mode, which affects performanc
...
+Remember that configuration files in `.mvn` are used when Maven is launched from 
+its parent directory; or when you build a _child_ project in multi-module build, the
+parent’s `.mvn` is used. For example, if you build `core`, `../.mvn` will be picked up because
+`core` is a child of `ejb-parent`; but if you build `ejb-parent`, `../.mvn` will not have any
+effect because `ejb-parent` is _not_ a child of `exonum-java-parent`.
+</details>
+
+See [Maven configuration documentation](https://maven.apache.org/configure.html) for details. 
+
+[build-benches]: https://jira.bf.local/browse/ECR-534?focusedCommentId=48501&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-48501
```
Comment (vitvakatu) [inline CONTRIBUTING.md]: I don't think it's really useful for others

Comment (dmitry-timofeev) [reply CONTRIBUTING.md]: This guide is not exclusively for "others", it is for (future) us just as well.

**Pull request merged**